runtime/vm/createramclass.cpp

```diff
@@ -2358,19 +2354,10 @@ internalCreateRAMClassDone(J9VMThread *vmThread, J9ClassLoader *classLoader, J9C
 			}
 		}
 
+		if (J9ROMCLASS_IS_VALUE(romClass)) {
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-		if (J9_ARE_ALL_BITS_SET(romClass->optionalFlags, J9_ROMCLASS_OPTINFO_IMPLICITCREATION_ATTRIBUTE)) {
-			U_16 implicitCreationFlags = getImplicitCreationFlags(romClass);
-			if (J9_ARE_ALL_BITS_SET(implicitCreationFlags, J9AccImplicitCreateNonAtomic)) {
-				classFlags |= J9ClassAllowsNonAtomicCreation;
-			}
-			if (J9_ARE_ALL_BITS_SET(implicitCreationFlags, J9AccImplicitCreateHasDefaultValue)) {
-				classFlags |= J9ClassAllowsInitialDefaultValue;
-			}
-		}
+			UDATA instanceSize = state->ramClass->totalInstanceSize;
```

Comment:
I would suggest getting rid of this variable and directly adding `state->ramClass->totalInstanceSize` in the if statement.